### PR TITLE
Don't check for free space on master folders (fixes #2236)

### LINKED ISSUE #2236: Minimum free disk space check should not run on master (read only) folders
Because we're not going to change any data there, so we don't care.

### PR DESCRIPTION
**Pull request by calmh**
Also clean up the logic a little, I thought it was a bit muddled.

### CODE REVIEW

File: lib/model/model.go [hunk 1]
```diff
@@ -1723,29 +1723,41 @@ func (m *Model) CheckFolderHealth(id string) error {
...
 			err = errors.New("folder marker missing")
-		} else if free, errDfp := osutil.DiskFreePercentage(folder.Path()); errDfp == nil && free < folder.MinDiskFreePct {
-			err = errors.New("out of disk space")
+
+		case !folder.ReadOnly:
+			// Check for free space, if it isn't a master folder. We aren't
+			// going to change the contents of master folders, so we don't
+			// care about the amount of free space there.
+			if free, errDfp := osutil.DiskFreePercentage(folder.Path()); errDfp == nil && free < folder.MinDiskFreePct {
+				err = errors.New("out of disk space")
```
Comment (AudriusButkevicius) [inline lib/model/model.go]: It's not actually out of space yet

Comment (calmh) [inline lib/model/model.go]: No, but that's the error we had before as well. :) What should it say, you think? "free space reserve not met"?

Comment (AudriusButkevicius) [inline lib/model/model.go]: About to run out of space, stopping.

Comment (arneko) [inline lib/model/model.go]: How about "insufficient disk space"?

**Comment by calmh** [comment]
Please take another look

**Pull request merged**